src/transform-inline/visitor-utils.ts

```diff
@@ -31,27 +31,33 @@ export function getPropertyInfo(symbol: ts.Symbol, visitorContext: VisitorContex
         throw new Error('Missing name in property symbol.');
     }
     if ('valueDeclaration' in symbol) {
-        if (!ts.isPropertySignature(symbol.valueDeclaration) && !ts.isMethodSignature(symbol.valueDeclaration)) {
-            throw new Error('Unsupported declaration kind: ' + symbol.valueDeclaration.kind);
+        const valueDeclaration = symbol.valueDeclaration as ts.PropertyDeclaration;
+        if (!ts.isPropertySignature(valueDeclaration)
+            && !ts.isMethodSignature(valueDeclaration)
+            && valueDeclaration.type !== undefined
+            && valueDeclaration.type.kind !== ts.SyntaxKind.FunctionType
```

Comment:
I'm wondering here why the check is needed:

```typescript
&& valueDeclaration.type !== undefined
&& valueDeclaration.type.kind !== ts.SyntaxKind.FunctionType
```

I tested it without and it seems to work fine. Is there a specific case I'm missing that requires these conditions?
Otherwise I would propose to just remove these two conditions, because then we can also drop the assertion `as ts.PropertyDeclaration` without any consequences.

Reply:
The condition is a leftover from my testing, but without the assertion I ran into build issues, see https://travis-ci.org/woutervh-/typescript-is/jobs/540501236 for example. I was wondering that you aren't getting these errors.